src/lib/comms/sol-oic-cbor.c

```diff
@@ -63,7 +63,7 @@ initialize_cbor_payload(struct sol_oic_map_writer *encoder)
 
     cbor_encoder_init(&encoder->encoder, encoder->payload,
         buf->capacity - offset, 0);
-    old_ptr = encoder->encoder.ptr;
+    old_ptr = encoder->encoder.data.ptr;
```

Comment:
I suppose that you mean cbor_encoder_get_extra_bytes_needed?
However, I think I will wait @glima for its expertise. The buffer_used_bump function and all its relative code seems to have been added by the commit 487af3523e230fa78dfe4a8dc00a3772adeda76e. It seems that soletta is "abusing" tinycbor API for a purpose that I do not master.